### PR TITLE
Refactor verification, integrity, parsing

### PR DESCRIPTION
**Pull request by decentralgabe**
Small refactors / moving / renames before implementing generic data integrity verification

### CODE REVIEW

File: credential/integrity/signature_test.go [hunk 1]
```diff
@@ -66,15 +64,15 @@ func TestVerifyCredentialSignature(t *testing.T) {
 		resolver, err := resolution.NewResolver([]resolution.Resolver{key.Resolver{}}...)
 		assert.NoError(tt, err)
 
-		credential := getTestCredential()
-		_, err = VerifyCredentialSignature(context.Background(), credential, resolver)
+		cred := getTestCredential()
+		_, err = VerifyCredentialSignature(context.Background(), cred, resolver)
 		assert.Error(tt, err)
-		assert.Contains(tt, err.Error(), "credential must have a proof")
+		assert.Contains(tt, err.Error(), "cred must have a proof")
```
Comment (andresuribe87) [inline credential/integrity/signature_test.go]: Is this intentional?

File: credential/integrity/signature_test.go [hunk 2]
```diff
@@ -66,15 +64,15 @@ func TestVerifyCredentialSignature(t *testing.T) {
...
 
 		// test with a pointer
-		_, err = VerifyCredentialSignature(context.Background(), &credential, resolver)
+		_, err = VerifyCredentialSignature(context.Background(), &cred, resolver)
 		assert.Error(tt, err)
-		assert.Contains(tt, err.Error(), "credential must have a proof")
+		assert.Contains(tt, err.Error(), "cred must have a proof")
```
Comment (andresuribe87) [inline credential/integrity/signature_test.go]: Here as well

File: credential/integrity/signature_test.go [hunk 3]
```diff
@@ -184,7 +182,7 @@ func TestVerifyJWTCredential(t *testing.T) {
 		jwtCred := getTestJWTCredential(tt, *signer)
 		_, err = VerifyJWTCredential(jwtCred, resolver)
 		assert.Error(tt, err)
-		assert.Contains(tt, err.Error(), "has no verification methods with kid: missing")
+		assert.Contains(tt, err.Error(), "has no validation methods with kid: missing")
```
Comment (andresuribe87) [inline credential/integrity/signature_test.go:185]: I think this should stay as verification

File: credential/status/statuslist2021.go [hunk 4]
```diff
@@ -250,7 +250,7 @@ func bitstringExpansion(compressedBitstring string) ([]string, error) {
 
 // ValidateCredentialInStatusList determines whether a credential is contained in a status list 2021 credential
 // https://w3c-ccg.github.io/vc-status-list-2021/#validate-algorithm
-// NOTE: this method does not perform credential signature/proof block verification
+// NOTE: this method does not perform credential signature/proof block validation
```
Comment (andresuribe87) [inline credential/status/statuslist2021.go:253]: verification?

File: credential/validation/validation.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+type (
+	// OptionKey uniquely represents an option to be used in a validator
+	OptionKey string
+)
+
+// Option represents a single option that may be required for a validator
+type Option struct {
+	ID     OptionKey
+	Option any
+}
```
Comment (andresuribe87) [inline credential/validation/validation.go:29]: Using `any` is discouraged unless there is a very good reason to do so. Can this be updated to received typed params?

File: credential/validation/validation.go
Comment (andresuribe87) [inline credential/validation/validation.go:1]: Why are we doing credential validation? I thought we should only be doing verification

File: cryptosuite/jws2020/jsonwebkey2020.go [hunk 6]
```diff
@@ -231,7 +231,7 @@ type JSONWebKeyVerifier struct {
 	jwx.Verifier
 }
 
-// Verify attempts to verify a `signature` against a given `message`, returning nil if the verification is successful
+// Verify attempts to verify a `signature` against a given `message`, returning nil if the validation is successful
```
Comment (andresuribe87) [inline cryptosuite/jws2020/jsonwebkey2020.go:234]: verification?

File: cryptosuite/bbs/bbsplussignaturesuite.go [hunk 7]
```diff
@@ -142,7 +142,7 @@ func (b BBSPlusSignatureSuite) Verify(v cryptosuite.Verifier, p cryptosuite.With
 	// make sure we set it back after we're done verifying
 	defer p.SetProof(proof)
 
-	// remove the proof value in the proof before verification
+	// remove the proof value in the proof before validation
```
Comment (andresuribe87) [inline cryptosuite/bbs/bbsplussignaturesuite.go:145]: verification?

File: cryptosuite/jws2020/jwssignaturesuite.go
Comment (andresuribe87) [inline cryptosuite/jws2020/jwssignaturesuite.go:1]: Should keep verification?

File: cryptosuite/jws2020/jwssignaturesuite_test.go [hunk 8]
```diff
@@ -153,7 +153,7 @@ func TestCredentialLDProof(t *testing.T) {
 		},
 	}
 
-	// create a copy for value verification later
+	// create a copy for value validation later
```
Comment (andresuribe87) [inline cryptosuite/jws2020/jwssignaturesuite_test.go:156]: seems like it shouldn't change

**Review by andresuribe87 — COMMENTED** [review]
Seems like multiple string replacements happened that shouldn't have happened. Can you confirm?

File: credential/integrity/signature_test.go (hunk 3 above)
Comment (decentralgabe) [reply credential/integrity/signature_test.go:185]: yes

File: credential/status/statuslist2021.go (hunk 4 above)
Comment (decentralgabe) [reply credential/status/statuslist2021.go:253]: yes

File: credential/validation/validation.go (hunk 5 above)
Comment (decentralgabe) [reply credential/validation/validation.go:29]: yes I have a separate ticket to refactor this #352

File: credential/validation/validation.go
Comment (decentralgabe) [reply credential/validation/validation.go:1]: we'll need both - for checking status, schema, etc

**Comment by decentralgabe** [comment]
@andresuribe87 should have addressed all

**Review by andresuribe87 — APPROVED** [review]

**Pull request merged**